pkg/operator/ceph/cluster/monitoring.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cluster to manage a Ceph cluster.
+package cluster
+
+import (
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mon"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/osd"
+)
+
+func (c *ClusterController) startClusterMonitoring(cluster *cluster, cephUser string) {
+	c.startCephMonMonitoring(cluster, cephUser)
+	c.startCephOSDMonitoring(cluster, cephUser)
+	c.startCephStatusMonitoring(cluster, cephUser)
+}
+
+func (c *ClusterController) startCephMonMonitoring(cluster *cluster, cephUser string) {
+	// Start the mon health checker
+	if monHealth, ok := cluster.monitoringChannels["mon"]; ok {
+		if monHealth.monitoringRunning {
+			// If the goroutine was running but the CR was updated to stop the monitoring we need to close the channel
+			if cluster.Spec.HealthCheck.DaemonHealth.Monitor.Disabled {
+				// close the channel so the goroutine can stop
+				close(cluster.monitoringChannels["mon"].stopChan)
```

Comment:
We are guaranteed to only close the channel once, right? Looks like the `monitoringRunning` ensures that since we change it back to false.

Reply:
Yes, we are safe, we won't close a closed channel, thanks to `monitoringRunning`.